miden-lib/asm/miden/tx.masm

```diff
@@ -160,3 +160,35 @@ export.build_recipient_hash
   swapw hmerge
   # [RECIPIENT]
 end
+
+#! Executes the provided procedure against the foreign account. 
+#!
+#! Inputs:  [foreign_account_id, FOREIGN_PROC_ROOT, ...]
+#! Outputs: [<values returned from the foreign procedure>]
+export.execute_foreign_procedure
+    # pad the stack before the syscall
+    push.0.0 movup.2
+    exec.kernel_proc_offsets::start_foreign_context_offset
+    padw swapw padw padw swapdw
+    # => [offset, foreign_account_id, pad(14), FOREIGN_PROC_ROOT, ...]
+
+    # load the foreign account to the memory
+    syscall.exec_kernel_proc
+    # => [pad(16), FOREIGN_PROC_ROOT, ...]
```

Comment:
One thing to consider (in a different PR) is whether we should change `start_foreign_context_offset` and `end_foreign_context_offset` into static kernel procedures (rather than dynamic ones). Let's create an issue for this.

Reply:
At first I created these procedures as static ones, but then thought that it will break the paradigm of dynamic kernel proc invocations and reworked them as dynamic procs. I don't yet understand why it will be better, but we can discuss this in the issue. 

Created issue: #894